naptime/src/main/scala/org/coursera/naptime/Types.scala

```diff
@@ -95,6 +98,27 @@ object Types extends StrictLogging {
           field.setProperties((properties ++ relatedMap).asJava)
       }
     }
+    for ((name, reverseRelation) <- fields.reverseRelations) {
+      Option(mergedSchema.getField(name)) match {
+        case Some(field) =>
+          logger.warn(s"Fields for resource $typeName tries to add reverse relation on field " +
+            s"'$name', but that field is already defined on the model")
+        case None =>
+          val errorMessageBuilder = new StringBuilder
+          val newField = new RecordDataSchema.Field(new ArrayDataSchema(new StringDataSchema)) // TODO(bryan): fix type here
```

Comment:
i believe it still works, because ids are serialized to strings in naptime output. I need to investigate this more though and try cases with integers and stuff. this is why i put the key type on the reverse relation annotation (`FinderReverseRelation[KeyType](...)`), but i don't know if that's necessary or if i can always assume the id will be a string.

i'm going to keep this todo in here and investigate when there's more time, but i don't think it'll break anything now.